lib/resources/shim-head.js

```diff
@@ -2,6 +2,25 @@
 ((win) => {
   win.ELECTRON = true;
 
+  if (process.env.ELECTRON_PROTOCOL_SERVE_INDEX && window.location.protocol !== 'file:') {
```

Comment:
@jacobq I get what you're saying, and perhaps further iterations down the road to make this more robust would make sense. But consider:

1. This logic only run's in Ember's `index.html` (because that's the only place we inject it)
1. It only has any effect if the renderer process accesses node modules
1. There's really no scenario where loading `index.html` from something other than a file protocol (whether literally `file:` or a custom protocol) *and* using node modules makes any kind of sense, or is not a gross violation of good security practices

So for those reasons I'm inclined to leave it as-is now, and if we run into any problems, we can iterate on it later. Since this file is in our addon, and not in the blueprint, making changes to it is easy and doesn't require the host app to modify any of their files.